Хеш и хеш-функции, множества  23/01/2023

ruslankorotkov opened the pull request.

ruslankorotkov committed:
Хеш и хеш-функции, множества  23/01/2023

ruslankorotkov committed:
Хеш и хеш-функции, множества  23/01/2023

ruslankorotkov committed:
Хеш и хеш-функции, множества  23/01/2023

dmitry-bizin commented on src/Main.java:
```diff
@@ -1,5 +1,168 @@
+
+        Set<Driver> driversSetList = new HashSet<>(Set.of(ivan, shuric, egor, nikita, artur, ilia, boris, nik, oleg, vova, zuric));
+        System.out.println(" множество водителей ");
+        for (Driver element : driversSetList) {
```
нужно через итератор вывести (см. задание)

ruslankorotkov committed:
Хеш и хеш-функции, множества  23/01/2023 (доделка)

ruslankorotkov committed:
Хеш и хеш-функции, множества  23/01/2023 (доделка)

dmitry-bizin commented on src/Main.java:
```diff
@@ -96,11 +96,19 @@ public static void main(String[] args) {
+        driversSetList.add(nikita);
+        driversSetList.add(artur);
+        driversSetList.add(ilia);
+        driversSetList.add(ilia);
```
это бесполезная проверка, потому что объект один и тот же. Даже если не переопределишь equals и hashCode, то при такой проверке они будут в единственном экземпляре, потому что один и тот же объект - для него equals и hashCode работают корректно по дефолту. Надо создать 2 разных объекта, но с одинаковыми значениями полей - и вот тут уже без переопределения equals и hashCode не обойтись

dmitry-bizin commented on src/Main.java:
```diff
@@ -1,5 +1,177 @@
+        driversSetList.add(ilia);
+
+        System.out.println(" множество водителей ");
+        driversSetList.iterator().hasNext();
```
здесь не совсем так, нужно использовать цикл, чтобы пройтись по элементам через итератор и вывести каждый

Commit:
Хеш и хеш-функции, множества  23/01/2023
 У меня в консоли пустота и надпись код 130.
 Множества как показано в сессии 26.01.2023 сделал.
 И ещё хочу сказать что я делаю git ,но что то из содержимого файла .gitignore : /.iml/   /.idea/   /out/
 кудато пропадает.

dmitry-bizin commented on src/Main.java:
```diff
@@ -104,12 +104,18 @@ public static void main(String[] args) {
+//        System.out.println(" множество водителей ");
+//        System.out.println(Arrays.toString(driversSetList.toArray()));
+        Iterator<Driver> iterD = driversSetList.iterator();
+        while (iterD.hasNext()) ;
```
точка с запятой не нужна после while, поэтому у тебя ничего и не выводится

dmitry-bizin commented on src/Main.java:
```diff
@@ -104,12 +104,18 @@ public static void main(String[] args) {
+//        System.out.println(Arrays.toString(driversSetList.toArray()));
+        Iterator<Driver> iterD = driversSetList.iterator();
+        while (iterD.hasNext()) ;
+        iterD.next().toString();
```
эта строка лишняя

Commit:
Хеш и хеш-функции, множества  23/01/2023  ( доработка )

Commit:
Хеш и хеш-функции, множества  23/01/2023  ( доработка )

Commit:
Хеш и хеш-функции, множества  23/01/2023  ( доработка )

Commit:
Хеш и хеш-функции, множества  23/01/2023  ( доработка )

Commit:
Хеш и хеш-функции, множества  23/01/2023  ( доработка )

Commit:
Хеш и хеш-функции, множества  23/01/2023  ( доработка )

Commit:
Хеш и хеш-функции, множества  23/01/2023  ( доработка )

dmitry-bizin commented on src/Main.java:
```diff
@@ -1,5 +1,183 @@
+        Set<Mechanic<?>> brigada3;
+        brigada3 = new HashSet<>(3);
+        brigada3.add(petr);
+        brigada3.add(misha);
```
здесь неверная проверка. Не надо один и тот же объект класть 2 раза, надо создать 2 РАЗНЫХ объекта, но с одинаковым наполнением и оба добавить во множество - это и будет правильная проверка, сейчас у тебя бесполезная проверка.

Commit:
Хеш и хеш-функции, множества  23/01/2023  ( доработка )
